### PR TITLE
Bump setuptools to 74.0.*

### PR DESCRIPTION
**Pull request by Avasam**
Closes #12598

### CODE REVIEW

File: stubs/setuptools/setuptools/command/easy_install.pyi [hunk 1]
```diff
@@ -112,6 +114,7 @@ class RewritePthDistributions(PthDistributions):
     prelude: str
     postlude: str
 
+def auto_chmod(func: Callable[..., _T], arg: str, exc: BaseException) -> _T: ...
```
Comment (Avasam) [inline stubs/setuptools/setuptools/command/easy_install.pyi:117]:
```suggestion
# Must match shutil._OnExcCallback
def auto_chmod(func: Callable[..., _T], arg: str, exc: BaseException) -> _T: ...
```

File: stubs/setuptools/pkg_resources/__init__.pyi [hunk 2]
```diff
@@ -428,6 +432,7 @@ class Distribution(NullProvider):
     def requires(self, extras: Iterable[str] = ()) -> list[Requirement]: ...
     def activate(self, path: list[str] | None = None, replace: bool = False) -> None: ...
     def egg_name(self) -> str: ...  # type: ignore[override]  # supertype's egg_name is a variable, not a method
+    def __getattr__(self, attr: str) -> Any: ...
```
Comment (srittau) [inline stubs/setuptools/pkg_resources/__init__.pyi:435]: Per our new policy to document any `Any` use: Could you add a comment as to why this is needed? (I think it forwards to another instance.)

File: stubs/setuptools/setuptools/errors.pyi [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
-LinkError = _distutils_errors.LinkError
-ModuleError = _distutils_errors.DistutilsModuleError
-OptionError = _distutils_errors.DistutilsOptionError
-PlatformError = _distutils_errors.DistutilsPlatformError
-PreprocessError = _distutils_errors.PreprocessError
-SetupError = _distutils_errors.DistutilsSetupError
-TemplateError = _distutils_errors.DistutilsTemplateError
-UnknownFileError = _distutils_errors.UnknownFileError
-BaseError = _distutils_errors.DistutilsError
+ByteCompileError: TypeAlias = _distutils_errors.DistutilsByteCompileError
```
Comment (srittau) [inline stubs/setuptools/setuptools/errors.pyi:5]: These don't look like type aliases to me, just like regular old runtime aliases.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, two notes below.

File: stubs/setuptools/pkg_resources/__init__.pyi (hunk 2 above)
Comment (Avasam) [reply stubs/setuptools/pkg_resources/__init__.pyi:435]: I think I can copy the docstring directly as a comment, it's clear enough.

**Review by srittau — APPROVED** [review]

**Pull request merged**